src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/CcdCallbackController.java

```diff
@@ -1,42 +1,52 @@
 package uk.gov.hmcts.reform.bulkscan.orchestrator.controllers;
 
+import com.google.common.collect.ImmutableList;
+import org.jetbrains.annotations.NotNull;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CallbackProcessor;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.AttachCaseCallbackService;
 import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
 import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
 import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
 
+import java.util.List;
+
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 import static org.springframework.http.ResponseEntity.ok;
 
 @RestController
 public class CcdCallbackController {
 
-    private final CallbackProcessor callbackProcessor;
+    private final AttachCaseCallbackService attachCaseCallbackService;
 
     @Autowired
-    public CcdCallbackController(CallbackProcessor callbackProcessor) {
-        this.callbackProcessor = callbackProcessor;
+    public CcdCallbackController(AttachCaseCallbackService attachCaseCallbackService) {
+        this.attachCaseCallbackService = attachCaseCallbackService;
     }
 
     @PostMapping(
-        path = "/callback/{type}",
+        path = "/callback/attach_case",
         consumes = APPLICATION_JSON_VALUE,
         produces = APPLICATION_JSON_VALUE
     )
-    public ResponseEntity<CallbackResponse> handleCallback(
-        @PathVariable("type") String type,
-        @RequestBody CallbackRequest callback) {
+    public ResponseEntity<CallbackResponse> handleCallback(@RequestBody CallbackRequest callback) {
+        if (callback != null && callback.getCaseDetails() != null) {
```

Comment:
This can be avoided by using built-in validation feature. Need to provide `@NotNull` and `@Valid` (maybe more) interfaces.

case details _nullability_ can be solved in service automatically too and necessary exception caught and translated in spring advice: proper and super clean way of dealing with this situation plus kind of minimising branching

Reply:
sure ... the hibernate validations would help here :D
but that's assuming you have control of the object definition being serialised.
Its in another repo so ... its debatable if we should add them there since it could break others.\
Maybe adding our own definition here might help.